core/src/main/java/hudson/scm/SCMDescriptor.java

```diff
@@ -56,7 +62,8 @@
      * @deprecated No longer used by default.
      */
     @Deprecated
-    public volatile int generation = 1;
+    @Restricted(NoExternalUse.class) @RestrictedSince("TODO")
+    public final AtomicInteger generation = new AtomicInteger(1);
```

Comment:
If we break compatibility anyway, why don't we make it private + add a proper getter method?

Reply:
@StefanSpieker WDYT?

Reply:
Sorry I missed that. Yes I will change that.